### DISCUSSION TITLE
Remove a delete method from models using extensions

### DISCUSSION THREAD

**Discussion by zomars** [prisma/prisma#17546] [Q&A] [answered]
[From the docs](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions#example-use-cases-for-extended-clients) I saw this use case but I struggle to find an actual example:

> Remove a delete method from models unless the application calls the admin endpoint and the user has the necessary privileges.

Any idea on how this looks in code? Can we just override with null? Or can we just throw an error if someone tries to use the method?

EDIT:

Tried this code so far:

```ts
  const xprisma = prisma.$extends({
    model: {
      deployment: {
        deleteMany() {
          throw new Error("Deployment shoudln't be deleted");
        },
      },
    },
  });
```

**Comment by nurul3101** [comment, marked as answer]
Hello @zomars 👋 

Here's an [example](https://github.com/sbking/prisma-client-extensions/blob/main/examples/readonly-client/script.ts) that you can use as a reference.

The example only allows certain methods to be accessed, you can tweak this example to allow all methods except the deleteMany method.

**Reply by zomars** [reply]
Nice and clean! Thank you so much 🙏🏽

**Reply by JonParton** [reply]
Just a heads up for others that the [Example link](https://github.com/prisma/prisma-client-extensions/blob/main/readonly-client/script.ts) has now moved

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
